pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -427,35 +429,31 @@ func (s *Server) createIstioCA(opts *caOptions) (*ca.IstioCA, error) {
 		// In Istiod, it is possible to provide one via "cacerts" secret in both cases, for consistency.
 		fileBundle.RootCertFile = ""
 	}
-	if _, err := os.Stat(fileBundle.SigningKeyFile); err != nil {
-		// The user-provided certs are missing - create a self-signed cert.
-		if s.kubeClient != nil {
-			log.Info("Use self-signed certificate as the CA certificate")
-
-			// Abort after 20 minutes.
-			ctx, cancel := context.WithTimeout(context.Background(), time.Minute*20)
-			defer cancel()
-			// rootCertFile will be added to "ca-cert.pem".
-			// readSigningCertOnly set to false - it doesn't seem to be used in Citadel, nor do we have a way
-			// to set it only for one job.
-			caOpts, err = ca.NewSelfSignedIstioCAOptions(ctx,
-				selfSignedRootCertGracePeriodPercentile.Get(), SelfSignedCACertTTL.Get(),
-				selfSignedRootCertCheckInterval.Get(), workloadCertTTL.Get(),
-				maxWorkloadCertTTL.Get(), opts.TrustDomain, true,
-				opts.Namespace, s.kubeClient.Kube().CoreV1(), fileBundle.RootCertFile,
-				enableJitterForRootCertRotator.Get(), caRSAKeySize.Get())
-		} else {
-			log.Warnf(
-				"Use local self-signed CA certificate for testing. Will use in-memory root CA, no K8S access and no ca key file %s",
-				fileBundle.SigningKeyFile)
+	if _, err := os.Stat(fileBundle.SigningKeyFile); err == nil {
+		detectedSigningCABundle = true
+	}
+	if _, err := os.Stat(path.Join(LocalCertDir.Get(), ca.IstioGenerated)); err == nil {
+		istioGenerated = true
+	}
+
+	if !detectedSigningCABundle || (detectedSigningCABundle && istioGenerated) {
+		if detectedSigningCABundle && istioGenerated {
+			log.Infof("%s secret found is IstioGenerated, use it as the CA certificate", ca.CACertsSecret)
 
-			caOpts, err = ca.NewSelfSignedDebugIstioCAOptions(fileBundle.RootCertFile, SelfSignedCACertTTL.Get(),
-				workloadCertTTL.Get(), maxWorkloadCertTTL.Get(), opts.TrustDomain, caRSAKeySize.Get())
+			// TODO(jaellio): Currently, istiod handles a "cacerts" secret with the "istio-generated" key the same way
+			// it handles the "istio-ca-secret" secret. Even though "cacerts" is file mounted, istiod will only watch the
+			// secret. If an "istio-ca-secret" exists in the control plane ns, it will be used instead of a "cacerts"
+			// secret with the "istio-generated" key.
+			// This will change in the future, and istiod will watch the file mount instead.
 		}
+
+		// Either the secret is not mounted, or it is mounted but the "istio-generated" key is used.
+		caOpts, err = s.createSelfSignedCACertificateOptions(&fileBundle, opts)
```

Comment:
This is under `!detectedSigningCABundle || (detectedSigningCABundle && istioGenerated)`

So does it mean it will self sign no matter it exists or not

Reply:
If a `cacerts` secret isn't mounted or if it's mounted and the `istio-generated` key is set, the secret should be created or loaded as a self-signed certificate (for step 1). The significance of this is that the `selfsignedcacertrotator` will be created rather than the cacerts file watcher.

Reply:
I donot think we should do duplicate work for this case `if it's mounted and the istio-generated key is set`.

And if we do run into self signing, we will fail later with conflict secret there.

Reply:
@hzxuzhonghu  Can you help me understand how this is duplicate work? We need istiod to do self-signing if istiod generate the certificate, so this step is necessary IMO

Reply:
> `if it's mounted and the istio-generated key is set.`

With this, the certificate is there, so that's confusing.  WHy don't we skip `createSelfSignedCACertificateOptions` immediately for this case

I might not capture the whole scene since this pr is only the  first part, nvm if I am mistaken

Reply:
> With this, the certificate is there, so that's confusing. WHy don't we skip createSelfSignedCACertificateOptions immediately for this case

According to https://github.com/istio/istio/pull/45291#issuecomment-1609003823, the plan is to move everything to use cacerts instead of istio-ca-secret. This PR is the first step in achieving that (in a non-breaking way) by adding an `istio-generated` field to cacerts to indicate that istiod owns the lifecycle of a cacerts secret as opposed to it being a user-created cacerts `Secret`. That's why 

> if it's mounted and the istio-generated key is set

the `createSelfSignedCACertificateOptions` function is called. Because the presence of the `istio-generated` key told us that it's an istiod owned secret and that we should enter the self signed cert code path

Reply:
Yes, ` it's an istiod owned secret`,
> `we should enter the self signed cert code path`

 but what does it do then?

Reply:
It configures the caOpts to be a self signed cert type instead of the regular plug-in cacerts code path. That caOpts gets consumed later when calling NewCA on L470